pyMBE.py

```diff
@@ -872,7 +874,49 @@ def create_counterions(self, object_name, cation_name, anion_name, espresso_syst
             for name in counterion_number.keys():
                 print(f'Ion type: {name} created number: {counterion_number[name]}')
         return counterion_number
+
+    def create_hydrogel(self, name, espresso_system):
+        """ 
+        creates the hydrogel `name` in espresso_system
+        Args:
+            name(`str`): Label of the hydrogel to be created. `name` must be defined in the `pmb.df`
+            espresso_system(`espressomd.system.System`): Instance of a system object from the espressomd library.
+
+        Returns:
+            hydrogel_info(`dict`):  {"name":hydrogel_name, "chains": {chain_id1: {residue_id1: {'central_bead_id': central_bead_id, 'side_chain_ids': [particle_id1,...]},...,"node_start":node_start,"node_end":node_end}, chain_id2: {...},...}, "nodes":{node1:[node1_id],...}}     
+        """
+        if not self.check_if_name_is_defined_in_df(name=name, pmb_type_to_be_defined='hydrogel'):
+            raise ValueError(f"Hydrogel with name '{name}' is not defined in the DataFrame.")
+        
+        hydrogel_info={"name":name, "chains":{}, "nodes":{}}
+        # placing nodes
+        node_positions = {}
+        node_topology = self.df[self.df["name"]==name]["node_map"].iloc[0]
+        if isinstance(node_topology, str):
+            node_topology = ast.literal_eval(node_topology) 
```

Comment:
@1234somesh why are lines 895-896 and the following needed?
```python
if isinstance(chain_topology, str):
            chain_topology = ast.literal_eval(chain_topology)
```
is it because  `chain_topology` and  `chain_topology` become strings when the `pmb.df` has been read from file? If yes, then we should modify `pmb.convert_columns_to_original_format` to parse  `chain_topology` and  `chain_topology` and transform them back to its original `dict` type. We can afterwards safely remove these lines from `pmb.create_hydrogel`

When parsing `chain_topology` and  `chain_topology`  in `pmb.convert_columns_to_original_format` please use    `json.loads()` instead than `ast.literal_eval()`, the former is a safer alternative that does the same job.